pytket/pytket/wasm/wasm.py

```diff
@@ -64,169 +72,181 @@ def __init__(self, filepath: str, check_file: bool = True, int_size: int = 32):
-                            ]
-                        ),
-                    )
+            if len(function_signatures[self._function_types[idx]]["return_types"]) > 1:
+                supported_function = False
+
+            if supported_function:
+                self._functions[x] = (
+                    len(
+                        function_signatures[self._function_types[idx]][
+                            "parameter_types"
+                        ]
+                    ),
+                    len(function_signatures[self._function_types[idx]]["return_types"]),
+                )
+
+            if not supported_function:
+                self._unsupported_functions.append(x)
 
-                if not supported_function:
-                    self._unsupported_function.append(x)
+        if "init" not in self._functions:
+            raise ValueError("wasm file needs to contain a function named 'init'")
 
-            if "init" not in self._functions:
-                raise ValueError("wasm file needs to contain a function named 'init'")
+        if self._functions["init"][0] != 0:
+            raise ValueError("init function should not have any parameter")
 
-            if self._functions["init"][0] != 0:
-                raise ValueError("init function should not have any parameter")
+        if self._functions["init"][1] != 0:
+            raise ValueError("init function should not have any results")
 
-            if self._functions["init"][1] != 0:
-                raise ValueError("init function should not have any results")
+        self.checked = True
+
+    @deprecated("Use public property checked instead.")
+    def _check_file(self) -> bool:
+        return self.checked
 
     def __str__(self) -> str:
-        """str representation of the wasm file"""
-        return self._wasmfileuid
+        """str representation of the wasm module"""
+        return self.module_uid
 
     def __repr__(self) -> str:
-        """str representation of the contents of the wasm file"""
-        if self._check_file:
-            result = f"Functions in wasm file with the uid {self._wasmfileuid}:\n"
-            for x in self._functions:
-                result += f"function '{x}' with "
-                result += f"{self._functions[x][0]} i{self._int_size} parameter(s)"
-                result += (
-                    f" and {self._functions[x][1]} i{self._int_size} return value(s)\n"
-                )
+        """str representation of the contents of the wasm file.
 
-            for x in self._unsupported_function:
-                result += (
-                    f"unsupported function with invalid "
-                    f"parameter or result type: '{x}' \n"
-                )
-
-            return result
-        else:
-            raise ValueError(
-                """the content of the wasm file can only be printed if the
-wasm file was checked"""
+        Will implicitly check the module if it has not been checked already.
```

Comment:
Not sure if this is desirable behavior, but it seemed more useful than just throwing an exception

Reply:
Yes, see the comment below. I think the problematic point is that this starts the type checking of all wasm functions that are added later on and it ignores that the user initially requested that they should not be checked.
Maybe it would be good to give a warning?